### DISCUSSION TITLE
Adding ability to split the runtime `token` function

### DISCUSSION THREAD

**Discussion by theMasix** [chakra-ui/panda#2193] [Ideas]
### Description

With regard to a design system tokens, the `token` function size (lines of code/bundle size) gets bigger and bigger.

Then If someone wants to use `token` only for `spacing` or `colors`, the `token` method should be imported completely (with a bunch of other unused values/variables)

It would be great if we can split the `token` method, OR  other `token` methods such as `spacingToken`, `colorToken` etc. become created by the Panda codegen.

### Problem Statement/Justification

Huge bundle size of the `token` method in case of a large design system token.

### Proposed Solution or API

Having the ability to split or generate multiple small token functions per token category.

**Comment by astahmer** [comment]
I think you could do that [with the hook](https://panda-css.com/docs/concepts/hooks) `codegen:prepare`

**Reply by theMasix** [reply]
Could you please provide an snippet for it?

**Reply by astahmer** [reply]
answering here as well, so that anyone finding this thread can benefit from it:

https://github.com/astahmer/pandabox/blob/f93c878e1dfaf41254f53bad034372f32d020a3a/packages/panda-plugins/src/minimal-setup.ts#L31
https://github.com/astahmer/pandabox/blob/f93c878e1dfaf41254f53bad034372f32d020a3a/packages/panda-plugins/src/strict-tokens-scope.ts#L29
https://github.com/astahmer/pandabox/blob/f93c878e1dfaf41254f53bad034372f32d020a3a/packages/panda-plugins/src/missing-css-warnings.ts#L16
https://github.com/astahmer/pandabox/blob/f93c878e1dfaf41254f53bad034372f32d020a3a/packages/panda-plugins/src/strict-tokens-runtime.ts#L30
